core/cas-server-core-services/src/test/java/org/apereo/cas/services/GroovyRegisteredServiceUsernameProviderTests.java

```diff
@@ -22,18 +25,23 @@
  * @since 5.2.0
  */
 @Tag("Groovy")
+@SpringBootTest(classes = {
+    RefreshAutoConfiguration.class,
+    CasCoreUtilConfiguration.class
+})
 public class GroovyRegisteredServiceUsernameProviderTests {
     private static final File JSON_FILE = new File(FileUtils.getTempDirectoryPath(), "GroovyRegisteredServiceUsernameProviderTests.json");
     private static final ObjectMapper MAPPER = JacksonObjectMapperFactory.builder()
         .defaultTypingEnabled(true).build().toObjectMapper();
 
     @Test
-    public void verifyUsernameProvider() {
+    public void verifyUsernameProvider() throws IOException {
         val p = new GroovyRegisteredServiceUsernameProvider();
-        p.setGroovyScript("file:///src/test/resources/uid.groovy");
+        val file = new File("./src/test/resources/uid.groovy");
```

Comment:
This version, as well as what was there before is somewhat suspicious. I think you should use `ClasspathResource` to load the groovy script, and use `ClasspathResource#getFile` to get it from there.

Reply:
The old file url was definitely wrong, I think I copied the absolute path stuff from another test, but I just changed it to `classpath:uid.groovy`.